pyproject.toml

```diff
@@ -69,7 +69,8 @@ recommended = [
     "matplotlib>=3.3,!=3.4.0,!=3.5.2",
 ]
 typing = [
-    "typing_extensions>=4.0.0"
+    "typing_extensions>=4.0.0",
+    "pandas-stubs>=1.4",
```

Comment:
This is adding a version requirement to `pandas-stubs`, but we don't have a version requirement for `pandas` itself: https://github.com/astropy/astropy/blob/769ad41652038d7d2aba9657a28f0ce8aaac982d/pyproject.toml#L84

Should we set a minimum supported version for `pandas` too and am I right in guessing that the `pandas-stubs` version matches the `pandas` version it is providing the stubs for?

Reply:
We should probably actually limit our support of Pandas to >2?  https://pypi.org/project/pandas/
@pllim, what do you think?

Reply:
According to the chart in https://scientific-python.org/specs/spec-0000/ , we still should support 1.5.0 until end of 2024?

Reply:
Oof. Ok! Then I can push a different PR for min pandas version.